benchmark_runner/common/clouds/BM/bare_metal_operations.py

```diff
@@ -0,0 +1,284 @@
+
+    def _get_kubeconfig(self):
+        """
+        This method returns kubeconfig if exist
+        :return:
+        """
+        if self._remote_ssh.exist(remote_path=self._provision_kubeconfig_path):
+            return self._remote_ssh.run_command(f'cat {self._provision_kubeconfig_path}')
+
+    def _wait_for_install_complete(self, sleep_time: int = SHORT_TIMEOUT):
+        """
+        This method waits till ocp install complete
+        :param sleep_time:
+        :return:
+        """
+        current_wait_time = 0
+        while self._provision_timeout <= 0 or current_wait_time <= self._provision_timeout:
+            install_log = self._remote_ssh.run_command(self._provision_installer_log)
+            if 'failed=0' in install_log:
+                return True
+            elif 'failed=1' in install_log:
+                return False
+            logger.info(f'Waiting till OCP install complete, waiting {int(current_wait_time/60)} minutes')
+            # sleep for x seconds
+            time.sleep(sleep_time)
+            current_wait_time += sleep_time
+        raise OCPInstallationFailed()
+
+    def _install_ocp_cmd(self):
+        """
+        This method returns local ocp installer cmd if exist
+        :return:
+        """
+        if self._remote_ssh.exist(remote_path=self._provision_installer_path):
+            return self._provision_installer_cmd
+
+    @logger_time_stamp
+    def connect(self):
+        """
+        This method connect to remote provision machine
+        :return:
+        """
+        self._remote_ssh.connect()
+
+    @logger_time_stamp
+    def disconnect(self):
+        """
+        This method
+        :return:
+        """
+        self._remote_ssh.disconnect()
+
+    def _get_installation_version(self):
+        """
+        This method return installation version
+        :return:
+        """
+        assisted_installer_versions = AssistedInstallerVersions()
+        if self.LATEST in self._install_ocp_version:
+            openshift_version_data = self._install_ocp_version.split('-')
+            # release candidate or engineering candidate or feature candidate version
+            if '-rc' in self._install_ocp_version or '-ec' in self._install_ocp_version or '-fc' in self._install_ocp_version:
+                return assisted_installer_versions.get_latest_version(latest_version=f'{openshift_version_data[1]}-{openshift_version_data[2]}')
+            # release version
+            else:
+                return assisted_installer_versions.get_latest_version(latest_version=openshift_version_data[1])
+        else:
+            return self._install_ocp_version
+
+    def get_ocp_server_version(self):
+        """
+        This method return ocp server version
+        :return:
+        """
+        return self._remote_ssh.run_command(command=f"{self._cli} version -ojson | jq -r '.openshiftVersion'").strip()
```

Comment:
Unfortunate that this doesn't take -ojsonpath, but nothing you can do about it.  Might want to add a comment that `-ojsonpath` doesn't work for `oc version` so nobody tries to change it later.